addons/pos_loyalty/models/pos_order.py

```diff
@@ -146,3 +147,29 @@ def _get_fields_for_order_line(self):
         fields = super(PosOrder, self)._get_fields_for_order_line()
         fields.extend(['is_reward_line', 'reward_id', 'coupon_id', 'reward_identifier_code', 'points_cost'])
         return fields
+
+    def _add_mail_attachment(self, name, ticket):
+        attachment = super()._add_mail_attachment(name, ticket)
+        gift_card_programs = self.config_id._get_program_ids().filtered(lambda p: p.program_type == 'gift_card')
+        if gift_card_programs:
+            gift_cards = self.env['loyalty.card'].search([('source_pos_order_id', '=', self.id),
+                                                          ('program_id', 'in', gift_card_programs.mapped('id'))])
+            if gift_cards:
+                for program in gift_card_programs:
+                    filtered_gift_cards = gift_cards.filtered(lambda gc: gc.program_id == program)
+                    if filtered_gift_cards:
+                        action_report = program.pos_report_print_id
```

Comment:
shouldn't we check that there is a 'pos_report_print_id' before using it?

Reply:
Robin made a fix where this become mandatory https://github.com/odoo/odoo/pull/102613

Reply:
oh nvm, I still need to check it since the settings can be on `scan_use`

Reply:
only in a specific configuration ( if self.gift_card_settings == "create_set":). If you use an existing card, you have no template.

Reply:
done in line 154